tests/unit/components/select-2-tags-test.js

```diff
@@ -0,0 +1,66 @@
+/*
+ Test tags including ember data support
+ */
+
+import Ember from "ember";
+import { test, moduleFor, moduleForComponent } from 'ember-qunit';
+import startApp from "../../helpers/start-app";
+
+var App, component;
+moduleForComponent('select-2', 'Select2Component (tags)', {
+  setup: function() {
+    App = startApp();
+    // setup and append component to dom
+    component = this.subject();
+  },
+  teardown: function() {
+    //Ember.run(App, 'destroy');
+    //Ember.run(component, 'destroy');
+  }
+});
+
+test("it can have ember data record array as content", function(assert) {
+  var store = App.__container__.lookup('store:main');
+
+  var post;
+
+  Ember.run(function() {
+    post = store.createRecord('post', {
+      title: "My post",
+      body: "Testing stuff"
+    });
+
+    var tag = store.createRecord('tag', {
+      name: "Ember.js"
+    });
+    post.get('tags').pushObject(tag);
+  });
+
+  component.setProperties({
+    tags: true,
+    value: post.get('tags'),
+    optionLabelPath: 'name'
+  });
+
+  this.render();
+
+  click('.select2-search-field');
+  fillIn('.select2-input', 'tag 2');
```

Comment:
I too had a problem with testing and filling in a search term which resulted in having to remove a complete test case in order to get 1.3.0 out of the door (the behaviour was correct when used manually, but I failed to test it correctly).

Maybe this can be resolved by scheduling the action of `options.query` (https://github.com/iStefo/ember-select-2/blob/master/addon/components/select-2.js#L160), but I am not sure about that :(